tv_apps/tv-deck/test/unit/tv_deck_test.js

```diff
@@ -0,0 +1,627 @@
+'use strict';
+/*jshint browser: true */
+/* global TVDeck, TVManager, TVTuner, TVSource, TVChannel */
+
+require('/shared/test/unit/mocks/mock_key_navigation_adapter.js');
+require('/shared/test/unit/mocks/mock_simple_key_navigation.js');
+require('/shared/test/unit/mocks/mock_tv_channel.js');
+require('/shared/test/unit/mocks/mock_tv_source.js');
+require('/shared/test/unit/mocks/mock_tv_tuner.js');
+require('/shared/test/unit/mocks/mock_tv_manager.js');
+require('/bower_components/smart-bubbles/script.js');
+require('/test/unit/mock_channel_manager.js');
+
+suite('tv-deck/tv_deck', function() {
+
+  var tvDeck;
+
+  function createMockElement(id, type) {
+    var element;
+    element = document.createElement(type || 'div');
+    element.id = id;
+    document.body.appendChild(element);
+    return element;
+  }
+
+  function createMockUI() {
+    createMockElement('overlay');
+    createMockElement('tv-stream', 'video');
+    createMockElement('channel-panel');
+    createMockElement('channel-number');
+    createMockElement('channel-title');
+    createMockElement('button-group-panel');
+    createMockElement('pin-button');
+    createMockElement('pin-button-contextmenu');
+    createMockElement('menu-button');
+    createMockElement('loading-icon');
+    createMockElement('bubble-animation', 'smart-bubbles');
+  }
+
+  suiteSetup(function(done) {
+    window.ChannelManager = window.MockChannelManager;
+    window.KeyNavigationAdapter = window.MockKeyNavigationAdapter;
+    window.SimpleKeyNavigation = window.MockSimpleKeyNavigation;
+    window.TVManager = window.MockTVManager;
+    window.TVTuner = window.MockTVTuner;
+    window.TVSource = window.MockTVSource;
+    window.TVChannel = window.MockTVChannel;
+
+    createMockUI();
+    sinon.stub(window, 'addEventListener');
+
+    require('/js/tv_deck.js', done);
+  });
+
+  suiteTeardown(function() {
+
+  });
+
+  suite('Constructor', function() {
+    setup(function() {
+      window.location.hash = '';
+      window.localStorage.removeItem('TV_Hash');
+      window.navigator.tv = new TVManager();
```

Comment:
You should save the original value before changing the window.navigator.tv. and set it back to orignal value at teardown().